src/Serialization/main.jl

```diff
@@ -188,57 +171,133 @@ function save_typed_object(s::SerializerState, x::T, key::Symbol) where T
   end
 end
 
+type_params(obj::T) where T = T, nothing
+
 function save_type_params(s::SerializerState, obj::Any, key::Symbol)
   set_key(s, key)
   save_type_params(s, obj)
 end
 
-function save_attrs(s::SerializerState, obj::T) where T
-  !with_attrs(s) && return 
-  if any(attr -> has_attribute(obj, attr), attrs_list(s, T))
-    save_data_dict(s, :attrs) do
-      for attr in attrs_list(s, T)
-        has_attribute(obj, attr) && save_typed_object(s, get_attribute(obj, attr), attr)
+save_type_params(s::SerializerState, obj::Any) = save_type_params(s, type_params(obj)...)
```

Comment:
```suggestion
save_type_params(s::SerializerState, obj::Any) = save_type_params(s, typeof(obj), type_params(obj)...)
```

I think this could do the trick

Reply:
yes this is probably fine

Reply:
actually, a little trickier than expected, there is an issue with the containers. 

I'll have to dig deeper, but I dont have time now

Reply:
I've pushed a fix for this now